src/main/core/main.rs

```diff
@@ -0,0 +1,258 @@
+    let config_file: ConfigFileOptions = serde_yaml::from_reader(file).context(format!(
+        "Could not parse configuration file {:?}",
+        &config_filename
+    ))?;
+
+    // generate the final shadow configuration from the config file and cli options
+    let config = ConfigOptions::new(config_file, options.clone());
+
+    if options.show_config {
+        eprintln!("{:#?}", config);
+        return Ok(());
+    }
+
+    // run any global C configuration handlers
+    unsafe { c::runConfigHandlers(&config as *const ConfigOptions) };
+
+    // disable log buffering during startup so that we see every message immediately in the terminal
+    shadow_logger::init().unwrap();
+    shadow_logger::set_buffering_enabled(false);
+
+    // start up the logging subsystem to handle all future messages
+    let log_level = config.general.log_level.as_ref().unwrap().to_c_loglevel();
+    unsafe { log_bindings::logger_setDefault(c::rustlogger_new()) };
+    unsafe { log_bindings::logger_setLevel(log_bindings::logger_getDefault(), log_level) };
+
+    // check if some log levels have been compiled out
+    let log_level = c_to_rust_log_level(log_level).unwrap();
+    if log_level > log::STATIC_MAX_LEVEL {
+        log::warn!(
+            "Log level set to {}, but messages higher than {} have been compiled out",
+            log_level,
+            log::STATIC_MAX_LEVEL,
+        );
+    }
+
+    // before we run the simluation, clean up any orphaned shared memory
+    unsafe { c::shmemcleanup_tryCleanup() };
+
+    // save the platform data required for CPU pinning
+    if config.experimental.use_cpu_pinning.unwrap() {
+        if unsafe { c::affinity_initPlatformInfo() } != 0 {
+            return Err(anyhow::anyhow!("Unable to initialize platform info"));
+        }
+    }
+
+    // raise fd soft limit to hard limit
+    raise_rlimit(libc::RLIMIT_NOFILE).context("Could not raise fd limit")?;
+
+    // raise number of processes/threads soft limit to hard limit
+    raise_rlimit(libc::RLIMIT_NPROC).context("Could not raise proc limit")?;
+
+    if config.experimental.use_sched_fifo.unwrap() {
+        set_sched_fifo().context("Could not set real-time scheduler mode to SCHED_FIFO")?;
+        log::debug!("Successfully set real-time scheduler mode to SCHED_FIFO");
+    }
+
+    // Disable address space layout randomization of processes forked from this
+    // one to improve determinism in cases when an executable under simulation
+    // branch on memory addresses.
+    disable_aslr().context("Could not disable plugin address space layout randomization")?;
+    log::debug!("ASLR disabled for processes forked from this parent process");
+
+    // check sidechannel mitigations
+    if unsafe { c::main_sidechannelMitigationsEnabled() } {
+        log::warn!(
+            "Speculative Store Bypass sidechannel mitigation is enabled (perhaps by seccomp?). \
+             This typically adds ~30%% performance overhead."
```

Comment:
I think this can just be a single `%` now, since it's not a C format string

Reply:
Fixed.